### PR TITLE
Add Habana tiles 

### PR DESCRIPTION
**Pull request by chen-levkovich**
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message.
- [ ] JIRA link(s):
- [ ] The Jira story is acked.
- [ ] Live build image: 
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [ ] QE contact acknowledges that this has been tested and is approved for merge.

### CODE REVIEW

**Review by lucferbux — CHANGES_REQUESTED** [review]
I'm gonna deploy it later to my cluster and I'll add screenshot so you can take a look @chen-levkovich 

In the meantime, I've spotted an issue, you need to add this newly created folder in the this kustomize file https://github.com/red-hat-data-services/odh-deployer/blob/main/odh-dashboard/apps/kustomization.yaml

**Comment by lucferbux** [comment]
Btw @zdtsw if we are doing some efforts to migrate ISV to odh-manifest, we should move the contribution files and so to have docs for external contributors who wants to add new info right?

**Comment by zdtsw** [comment]
> Btw @zdtsw if we are doing some efforts to migrate ISV to odh-manifest, we should move the contribution files and so to have docs for external contributors who wants to add new info right?

I think the final goal is to not use "odh-deployer" for ISV. 
Till new operator is stable, we still need make the double efforts to keep add/update ISV in odh-deployer (i.e for Habana) and sync such into odh-manifasts. 
But for the current/ongoing release, I do not think the plan is to have Habana for new operator (just to mitigate risk from too much integration work)

**Comment by harshad16** [comment]
/hold 

Holding the PR for getting some clarity on this inclusion.
@chen-levkovich  could please include the description.

**Comment by chen-levkovich** [comment]
What description should I include?

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 1]
```diff
@@ -0,0 +1,245 @@
+apiVersion: console.openshift.io/v1
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:1]: This is the actual output:

```bash
The OdhQuickStart "pytorch-tutorial-habana-ai" is invalid: 
* spec.tasks[0].summary.failed: Required value
* spec.tasks[0].summary.success: Required value
* spec.tasks[0].review.instructions: Invalid value: "": spec.tasks[0].review.instructions in body should be at least 1 chars long
* spec.tasks[1].summary.success: Required value
* spec.tasks[1].review.instructions: Invalid value: "": spec.tasks[1].review.instructions in body should be at least 1 chars long
* spec.tasks[2].review.instructions: Invalid value: "": spec.tasks[2].review.instructions in body should be at least 1 chars long
* spec.tasks[2].summary.success: Required value
```

File: odh-dashboard/apps/habana/habana_kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana_kustomization.yaml:1]: This file should be name just `kustomization`

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+        if __name__ == '__main__':
+            main()
+        ```
+      review:
+        instructions: |-
+
+        failedTaskHelp: This task is not verified yet. Try the task again.
+      summary:
+        success:
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:185]: You need succcess or remove the field

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+        ### Lines 47, 52 - In Lazy mode, mark_step() must be added in all training scripts right after loss.backward() and optimizer.step().
+        ```
+        htcore.mark_step()
+        ```
+      review:
+        instructions: |-
+          
+        failedTaskHelp: This task is not verified yet. Try the task again.
+      summary:
+        success:
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:207]: And here too

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        ### Line 104 - Target the Gaudi HPU device:
+        ```
+        device = torch.device("hpu")
+        ```
+        ### Lines 47, 52 - In Lazy mode, mark_step() must be added in all training scripts right after loss.backward() and optimizer.step().
+        ```
+        htcore.mark_step()
+        ```
+      review:
+        instructions: |-
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:203]: You need some value here

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        469/469 [==============================] - 1s 2ms/step - loss: 0.5845 - accuracy: 0.8606
+        Epoch 4/5
+        469/469 [==============================] - 1s 2ms/step - loss: 0.5237 - accuracy: 0.8688
+        Epoch 5/5
+        469/469 [==============================] - 1s 2ms/step - loss: 0.4865 - accuracy: 0.8749
+        313/313 [==============================] - 1s 2ms/step - loss: 0.4482 - accuracy: 0.8869
+        ```
+        Since the first iteration includes graph compilation time, you can see the first iteration takes longer to run than later iterations. The software stack compiles the graph and saves the recipe to cache. Unless the graph changes or a new graph comes in, no recompilation is needed during the training. Typically, the graph compilation happens at the beginning of the training and at the beginning of the evaluation.
+      review:
+        instructions: |-
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:237]: Same here

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+         
+        failedTaskHelp:
+          This task is not verified yet. Try the task again.
+      summary:
+        success:
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:242]: Same here

File: odh-dashboard/apps/habana/habana-app.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+apiVersion: dashboard.opendatahub.io/v1
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-app.yaml:1]: You'll need to add this

`  - ./habana`

At the end of `oah-dashboard/apps/kustomization.yaml`

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+
+                torch.save(net.state_dict(), os.path.join(save_path,'epoch_{}.pth'.format(epoch)))
+
+                scheduler.step()
+
+        if __name__ == '__main__':
+            main()
+        ```
+      review:
+        instructions: |-
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:181]: And here too

**Review by lucferbux — CHANGES_REQUESTED** [review]

**Comment by lucferbux** [comment]
@chen-levkovich here you have a few screenshots:
Screenshot 2023-07-28 at 10 37 17
Screenshot 2023-07-28 at 10 36 59

**Comment by lucferbux** [comment]
btw @taneem-ibrahim @Jooho 

Can you kindly take over this? I'm still not sure who I need to contact regarding this kind of situations. We are always willing to help here, but I would love to know the actual flow to be more efficient the next time.

Thanks in advance.

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+                torch.save(net.state_dict(), os.path.join(save_path,'epoch_{}.pth'.format(epoch)))
+
+                scheduler.step()
+
+        if __name__ == '__main__':
+            main()
+        ```
+      review:
+        failedTaskHelp: This task is not verified yet. Try the task again.
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:181]: You still need the `instructions` attribute, try something like this:

```suggestion
        instructions: |-
          #### Verify the file is created:
          Do you have a file called `example.py`?
        failedTaskHelp: This task is not verified yet. Try the task again.
```

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+        ### Line 104 - Target the Gaudi HPU device:
+        ```
+        device = torch.device("hpu")
+        ```
+        ### Lines 47, 52 - In Lazy mode, mark_step() must be added in all training scripts right after loss.backward() and optimizer.step().
+        ```
+        htcore.mark_step()
+        ```
+      review:
+        failedTaskHelp: This task is not verified yet. Try the task again.
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:201]: Same here, try to add something like this:

```suggestion
        instructions: |-
          #### Verify the lines are present in the file:
          Do you have the lines in the file?
        failedTaskHelp: This task is not verified yet. Try the task again.
```

File: odh-dashboard/apps/habana/habana-quickstart.yaml [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+        Epoch 4/5
+        469/469 [==============================] - 1s 2ms/step - loss: 0.5237 - accuracy: 0.8688
+        Epoch 5/5
+        469/469 [==============================] - 1s 2ms/step - loss: 0.4865 - accuracy: 0.8749
+        313/313 [==============================] - 1s 2ms/step - loss: 0.4482 - accuracy: 0.8869
+        ```
+        Since the first iteration includes graph compilation time, you can see the first iteration takes longer to run than later iterations. The software stack compiles the graph and saves the recipe to cache. Unless the graph changes or a new graph comes in, no recompilation is needed during the training. Typically, the graph compilation happens at the beginning of the training and at the beginning of the evaluation.
+      review:
+        failedTaskHelp:
+          This task is not verified yet. Try the task again.
```
Comment (lucferbux) [inline odh-dashboard/apps/habana/habana-quickstart.yaml:234]: And finally here:

```suggestion
        instructions: |-
          #### Verify the output is as expected:
          Do you see the expected output?
        failedTaskHelp:
          This task is not verified yet. Try the task again.
```

**Review by lucferbux — CHANGES_REQUESTED** [review]
There's only one change left, I've applied the suggestions I'm asking bellow and with that you can see that is working:

Screenshot 2023-09-05 at 12 34 22

**Comment by zdtsw** [comment]
@lucferbux can we close this one if the change has been merged into odh-dashboard?

**Comment by lucferbux** [comment]
It came on hold, I'm not sure if @chen-levkovich wanted to continue with the introduction of the habana tile
